### PR TITLE
feature: add new event ResultTestVariablesAfterTransmissionEvent

### PR DESCRIPTION
**Pull request by Karol-Stelmaczonek**
Ticket: https://oat-sa.atlassian.net/browse/INF-235

## What's Changed

- add new event ResultTestVariablesAfterTransmissionEvent

https://github.com/oat-sa/extension-tao-testqti/assets/118974926/7c1f4265-0ed8-426f-baab-7abcd30f106a

### CODE REVIEW

**Comment by pribi** [comment]
Can you please add test coverage to the new code created?

File: helpers/class.TestSession.php [hunk 1]
```diff
@@ -814,14 +819,49 @@ private function triggerResultTestVariablesTransmissionEvent(
...
             $testUri
+        );
+
+        $this->getEventManager()->attach($event, [$this, 'attachToTransmissionEvent']);
+        $this->getEventManager()->trigger($event);
+    }
+
+    public function attachToTransmissionEvent(ResultTestVariablesTransmissionEvent $event): void
+    {
+        $scoreTotal = array_filter($event->getVariables(), function ($item) {
```
[the remark this answers is not in the document]
Comment (Karol-Stelmaczonek) [reply helpers/class.TestSession.php:835]: Done, thanks

File: helpers/class.TestSession.php [hunk 2]
```diff
@@ -814,14 +819,49 @@ private function triggerResultTestVariablesTransmissionEvent(
...
+            $this->isManualScored()
         ));
     }
 
+    private function isManualScored(): bool
+    {
+        /** @var AssessmentItemRef $itemRef */
+        foreach ($this->getRoute()->getAssessmentItemRefs() as $itemRef) {
+            foreach ($itemRef->getComponents() as $component) {
+                if ($component instanceof OutcomeDeclaration) {
```
[the remark this answers is not in the document]
Comment (Karol-Stelmaczonek) [reply helpers/class.TestSession.php:854]: Thanks, done!

**Review by pnal — CHANGES_REQUESTED** [review]

File: helpers/class.TestSession.php [hunk 3]
```diff
@@ -814,14 +819,54 @@ private function triggerResultTestVariablesTransmissionEvent(
...
     ): void {
-        $this->getEventManager()->trigger(new ResultTestVariablesTransmissionEvent(
+        $event = new ResultTestVariablesTransmissionEvent(
             $this->getSessionId(),
             $variables,
             $this->getSessionId(),
             $testUri
+        );
+
+        $this->getEventManager()->attach($event, [$this, 'attachToTransmissionEvent']);
```
[the remark this answers is not in the document]
Comment (andreluizmachado) [reply helpers/class.TestSession.php:829]: The listeners of this event has to be registered via migrations

[the remark this answers is not in the document]
Comment (Karol-Stelmaczonek) [reply helpers/class.TestSession.php]: I've rewritten the whole event logic here and it follows yours suggestions now.

**Review by pnal — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [x] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

File: models/classes/eventHandler/ResultTransmissionEventHandler/ResultTransmissionEventHandler.php [hunk 4]
```diff
@@ -64,4 +78,59 @@ private function buildTransmitter($deliveryExecutionId): taoQtiCommon_helpers_Re
...
+            }
+        );
+    }
+
+    /**
+     * @param ResultTestVariablesTransmissionEvent $event
+     * @return void
+     * @throws InvalidServiceManagerException
+     */
+    private function triggerTestVariablesRecorded(ResultTestVariablesTransmissionEvent $event): void
```
Comment (andreluizmachado) [inline models/classes/eventHandler/ResultTransmissionEventHandler/ResultTransmissionEventHandler.php:127]: return type void will be deprecated on php8.1

Comment (andreluizmachado) [reply models/classes/eventHandler/ResultTransmissionEventHandler/ResultTransmissionEventHandler.php:127]: ok, just when Returning by reference

File: models/classes/eventHandler/ResultTransmissionEventHandler/ResultTransmissionEventHandler.php [hunk 5]
```diff
@@ -64,4 +78,59 @@ private function buildTransmitter($deliveryExecutionId): taoQtiCommon_helpers_Re
...
+        }
+
+        return $storage;
+    }
+
+    /**
+     * @param ResultTestVariablesTransmissionEvent $event
+     * @return array
+     */
+    private function containsScoreTotal(ResultTestVariablesTransmissionEvent $event): array
```
[the remark this answers is not in the document]
Comment (Karol-Stelmaczonek) [reply models/classes/eventHandler/ResultTransmissionEventHandler/ResultTransmissionEventHandler.php:112]: Yes thats good point, changed

**Review by andreluizmachado — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful
- [x] Pull request's target is not `master`

**Review by pribi — APPROVED** [review]
LGTM, but not tested.

**Pull request merged**